framework/db/ActiveQueryTrait.php

```diff
@@ -129,12 +129,14 @@ private function createModels($rows)
             if ($this->indexBy === null) {
                 foreach ($rows as $row) {
                     $model = $class::instantiate($row);
+                    $class = get_class($model);
```

Comment:
This is problematic as you are modifying `$class` which would be used in the next loop.